src/server/server_dispatch.c

```
@@ -1178,144 +1237,105 @@ static void handle_game_message(int peer_slot, const bc_transport_msg_t *msg)
+            peer->object_id = bc_make_ship_id(game_slot);
+        }
+
+        /* Initialize server-side ship state from the ship blob */
+        if (g_registry_loaded && opcode == BC_OP_OBJ_CREATE_TEAM &&
+            have_ship_blob) {
+            int cidx = bc_registry_find_ship_index(&g_registry,
+                                                   bhdr.species_id);
+            if (cidx >= 0) {
+                const bc_ship_class_t *cls =
+                    bc_registry_get_ship(&g_registry, cidx);
+                u8 team_id = (header_ok && hdr.has_team) ? hdr.team_id : 0;
```

Clang (14.0.6)

[note] +4, including nesting penalty of 3, nesting level increased to 4

(clang)